File: ruby/series/series.rb
class Series
  def initialize(num)
    @num = num
  end

  def slices(n)
    if n > @num.length
      raise ArgumentError.new("Cannot have a series greater than the length")
    end

    series_arr = []
    num_arr = @num.split("").map {|num| num.to_i }
    index = 0
    while index < num_arr.length
      series_arr << num_arr.slice(index...(index+n))
      index += 1
    end
    series_arr.delete_if {|series|
      series.length != n
    }
  end
end

File: javascript/series/series.js
var Series = function(str){
  this.digits = this.digitize(str);
}

Series.prototype.digitize = function(str){
  return str.split("").map(function(digit){
    return parseInt(digit);
  });
}

Series.prototype.slices = function(sliceSize){
  var digits = this.digits;

  // not enough digits to slice
  if(digits.length < sliceSize){
    throw 'Slice size is too big.';
  }

  var seriesArr = [];

  for(var i = 0; i < digits.length; i++){
    if(digits.slice(i, i + sliceSize).length === sliceSize){
      seriesArr.push(digits.slice(i, i + sliceSize));
    }
  }

  return seriesArr;
}

module.exports = Series;
